src/transformers/models/gpt_neox/modeling_gpt_neox.py

```diff
@@ -891,6 +943,27 @@ def forward(
         if inputs_embeds is None:
             inputs_embeds = self.embed_in(input_ids)
 
+        # Attention mask.
+        if attention_mask is not None:
+            assert batch_size > 0, "batch_size has to be defined and > 0"
+            attention_mask = attention_mask.view(batch_size, -1)
+            if self._use_flash_attention_2:
+                attention_mask = attention_mask if 0 in attention_mask else None
+            elif self._use_sdpa and not output_attentions and not any(head_mask):
```

Comment:
Added head mask check as `not any(head_mask)` since it's converted to `[None] * self.config.num_hidden_layers` in the calls above.

Reply:
Changed this to a simplified check by moving the `head_mask` creation to a later point.